cores/esp8266/IPAddress.cpp

```diff
@@ -63,52 +74,84 @@ bool IPAddress::fromString(const char *address) {
                 // Too much dots (there must be 3 dots)
                 return false;
             }
-            _address.bytes[dots++] = acc;
+            (*this)[dots++] = acc;
             acc = 0;
         }
         else
         {
             // Invalid char
+#if LWIP_IPV6
+            return fromString6(org);
+#else
             return false;
+#endif
         }
     }
 
     if (dots != 3) {
         // Too few dots (there must be 3 dots)
         return false;
     }
-    _address.bytes[3] = acc;
+    (*this)[3] = acc;
+
+    setV4();
     return true;
 }
 
 IPAddress& IPAddress::operator=(const uint8_t *address) {
-    memcpy(_address.bytes, address, sizeof(_address.bytes));
+    setV4();
```

Comment:
E.g.:
```
IPAddress& IPAddress::operator=(const uint16_t *address)
```

Reply:
There is `raw_address()` above that we discussed about, that we keep because it is Arduino API. I just realized I added `uint16_t* raw6()` because that was needed at least by the print function (modulo endianness that we will check later with compilation on host - at least it works on the ESP).
So we almost have it, but I keep thinking it should not get too much exposed because I saw nobody messing with v6 addresses in binary format. I did not get deeply into RFCs but I guess it is stated somewhere that users should stay away from this.